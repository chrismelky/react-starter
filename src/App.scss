@import '_variables.scss';

html {
  font-size: 16px;
}

html,
body {
  height: 100%;
}

body {
  margin: 0;
  font-family: Roboto, 'Helvetica Neue', sans-serif;
  font-size: 12px;
  color: #202124;
}

section {
  background-color: $contentBg;
}

.p-sidebar-left .p-sidebar {
  /* box-shadow:  0 11px 15px -7px rgb(0 0 0 / 20%), 0 24px 38px 3px rgb(0 0 0 / 14%), 0 9px 46px 8px rgb(0 0 0 / 12%); */
  box-shadow: 0 3px 1px -2px rgb(0 0 0 / 20%), 0 2px 2px 0px rgb(0 0 0 / 14%),
    0 1px 5px 0px rgb(0 0 0 / 12%);
  width: 256px;
  background-color: $navBg;
  color: #fff;
}

.sidebar {
  width: 21rem;
  overflow: hidden;
  transform: translate3d(0px, 0px, 0px);
  transition: all 0.3s;
}
.p-sidebar .p-sidebar-content {
  padding: $navPadding;
}
.p-sidebar-left .p-sidebar-enter {
  box-shadow: none !important;
}

.p-component {
  font-size: 14px;
}

.p-card {
  // border-radius: 0.5rem;

  .p-card-body {
    padding: 0.75rem;
  }
  .p-card-content {
    padding: 0;
  }
}
.p-dialog {
  .p-dialog-header {
    padding: 1rem 1.5rem;
  }
  .p-dialog-content {
    padding-top: 1rem;
  }
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.p-toolbar {
  padding: 0.5rem;
  background-color: #fff;
  border: none;
}

.p-panelmenu .p-panelmenu-header {
  > a {
    border: none !important;
    .p-menuitem-icon {
      margin-right: 0.7rem;
    }
  }
}

.p-panelmenu .p-panelmenu-content {
  border: none !important;
  // background-color: transparent !important;
}
.p-panelmenu .p-panelmenu-header.p-highlight {
  border-top-right-radius: 0.375rem;
  border-top-left-radius: 0.375rem;
}

.p-panelmenu .p-panelmenu-panel .p-panelmenu-header .p-panelmenu-icon {
  font-size: 14px;
}

.sidenav-item {
  padding: 0px;
  margin-top: 32px;
  background-color: transparent;

  .p-panelmenu {
    width: 100%;
    background-color: transparent;

    .p-menuitem-icon {
      font-size: 18px;
      font-weight: 500;
      color: $navIcon;
    }

    .pi {
      font-weight: 500;
    }

    .pi-chevron-right {
      font-size: $fontSize;
    }

    .p-menuitem-text {
      font-size: 14px;
      line-height: 18px;
      font-weight: 400;
      overflow-x: hidden;
    }

    .p-panelmenu-content {
      background-color: rgba(255, 255, 255, 0.4);
      color: $navText;
      padding-left: 0.75rem;

      .p-panelmenu-icon {
        display: none;
      }

      .p-menuitem {
        .p-menuitem-link {
          .p-menuitem-icon,
          .p-menuitem-text {
            color: $navText;
          }
        }

        .p-menuitem-link:hover {
          .p-menuitem-icon,
          .p-menuitem-text {
            color: white;
          }
        }

        .p-menuitem-link-active {
          .p-menuitem-icon,
          .p-menuitem-text {
            color: #f3a71f;
            line-height: 18px;
          }
        }
      }
    }

    .p-highlight {
      background-color: $navHiglight;
    }

    li.p-menuitem:hover {
      background-color: transparent !important;
    }

    .p-panelmenu-panel:first-child .p-panelmenu-header > a {
      background-color: transparent;
      color: $navText;
    }

    .p-panelmenu-panel .p-panelmenu-header > a {
      color: $navText;
      background-color: transparent;
    }

    .p-panelmenu-panel {
      box-shadow: none;
    }
  }
}

.p-datatable .p-datatable-header {
  background: transparent;
  border: none;
}

.p-datatable {
  .p-datatable-thead > tr > th,
  .p-sortable-column.p-highlight {
    background: $contentBg;
  }
}

.p-datatable.p-datatable-sm .p-datatable-tbody > tr > td {
  padding: 0.125rem 0.5rem;
  font-size: 13px;
}

.p-paginator {
  .p-button:disabled {
    background-color: transparent !important;
  }

  .p-paginator-first,
  .p-paginator-prev,
  .p-paginator-next,
  .p-paginator-last {
    border: none !important;
  }

  .p-paginator-pages {
    .p-paginator-page.p-highlight {
      background: #4f46e5;
      border-radius: 0.375rem;
      border: 1px solid #4f46e5;
      color: #fff;
    }
    .p-paginator-page {
      border: none;
      min-width: 2rem;
      height: 2rem;
    }
  }
}
.p-button-label,
span.p-column-title,
.p-sortable-column-icon,
.p-column-filter .pi-filter-icon {
  text-transform: uppercase;
  font-size: 12px;
}
td.actions {
  width: 100px;
  text-align: right;
}

.p-paginator .p-paginator-current {
  margin-left: auto;
}

.p-panelmenu .no-child .p-panelmenu-content {
  // padding: 0 !important;
  display: none !important;
}

.sidenav-item .p-panelmenu .no-child .p-highlight {
  border-bottom: none !important;
}

.p-column-filter-menu {
  margin-left: initial !important;
}
