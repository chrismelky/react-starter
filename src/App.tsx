import React, { lazy, useEffect } from 'react';

import 'primereact/resources/primereact.min.css';
import 'primereact/resources/themes/mdc-light-indigo/theme.css'; //theme                 //core css
import 'primeicons/primeicons.css';
import 'primeflex/primeflex.css';
import './app.scss'; //icons
import { Route, Routes } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { getSession } from './reducers/authentication';
import Main from './modules/layout/main';
import Login from './modules/login/login';
import PrivateRouteComponent from './private-route';

const Dashboard = lazy(() => import('./modules/dashboard'));
const UserList = lazy(() => import('./modules/user'));
const FacilityTypeList = React.lazy(() => import('./modules/facility-type'));
/* ====Chrispro lazy component Generator Hook: Dont Delete==== */

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getSession());
    //eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Routes>
      <Route path="/" element={<Main />}>
        <Route
          path="/"
          element={<PrivateRouteComponent component={Dashboard} />}
        />
        <Route
          path="/user"
          element={
            <PrivateRouteComponent
              component={UserList}
              hasAnyAuthorities={[]}
            />
          }
        />

        <Route
          path="/facility-type"
          element={
            <PrivateRouteComponent
              component={FacilityTypeList}
              hasAnyAuthorities={[]}
            />
          }
        />
        {/* ====Chrispro router Generator Hook: Dont Delete==== */}
      </Route>
      <Route path="/login" element={<Login />} />
    </Routes>
  );
}

export default App;
