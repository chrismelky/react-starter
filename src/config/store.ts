import { configureStore } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import authenticationReducer from '../reducers/authentication';

export const store = configureStore({
  reducer: {
    authentication: authenticationReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export const useAppDispatch = () => useDispatch<AppDispatch>();
