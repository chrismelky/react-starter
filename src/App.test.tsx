import React from 'react';
import App from './app';
import { store } from './config/store';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { QueryClient, QueryClientProvider } from 'react-query';
import renderer from 'react-test-renderer';

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
      retry: false,
    },
  },
});

describe('App Test', () => {
  test('Render Correct', () => {
    const tree = renderer
      .create(
        <QueryClientProvider client={queryClient}>
          <Provider store={store}>
            <MemoryRouter>
              <App />
            </MemoryRouter>
          </Provider>
        </QueryClientProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
