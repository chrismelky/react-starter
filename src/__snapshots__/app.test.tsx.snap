// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Render Correct 1`] = `
<div>
  Connecting....
</div>
`;
